src/praisonai-agents/tests/test_self_reflection_comprehensive.py

```
@@ -0,0 +1,117 @@
+    try:
+        # Start execution
+        result = agents.start()
+        print(f"Result with tools: {result}")
+        
+        assert result, "Self-reflection with tools failed to produce a result."
+        print("\n✅ SUCCESS: Self-reflection with tools is working!")
+        return result
+        
+    except Exception as e:
+        print(f"\n❌ ERROR: {str(e)}")
+        raise AssertionError(f"Test with tools failed: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)